apiary.apib

```diff
@@ -1738,6 +1738,699 @@ root will not be listable, as it is virtual. Access known paths directly.
+
+        ```
+        ERROR: Please specify a version to install
+        ```
+
++ Request With Invalid Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/update/install?arg=kitten"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/update/install?arg=kitten"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: text/plain
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Stream-Output: 1
+        Date: Thu, 11 Feb 2016 22:26:07 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        fetching ipfs version vkitten
+        install failed, reverting changes...
+        ERROR: no link named "vkitten" under QmdvSvgDA2C64eGcRKnWiBwZCdM6WCKSMa38d1hsvMkcRZ 
+        ```
+
++ Request With Valid Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/update/install?arg=v0.3.5"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/update/install?arg=v0.3.5"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: text/plain
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Stream-Output: 1
+        Date: Thu, 11 Feb 2016 22:26:07 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        fetching ipfs version v0.3.5
+        binary downloaded, verifying...
+        install failed, reverting changes...
+        ERROR: version didnt match 
+        ```
+
+## stash [GET /update/stash]
+Stashes copy of currently installed IPFS binary.
+
++ Request Without Arguments
+
+    #### curl
+
+        curl -i "http://localhost:5001/api/v0/update/stash"
+
+    + Body
+
+        ```
+        curl -i "http://localhost:5001/api/v0/update/stash"
+        ```
+
++ Response 200
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
+        Access-Control-Expose-Headers: X-Stream-Output, X-Chunked-Output
+        Content-Type: text/plain
+        Server: go-ipfs/0.4.0-dev
+        Trailer: X-Stream-Error
+        Transfer-Encoding: chunked
+        X-Stream-Output: 1
+        Date: Thu, 11 Feb 2016 22:28:02 GMT
+        Transfer-Encoding: chunked
+        ```
+
+    + Attributes (string)
+
+    + Body
+
+        ```
+        ```
+
+## revert [GET /update/revert{?arg}]
+Revert to previously installed version of IPFS.
+
++ Request Without Arguments
+
+    The response is the same if the argument is empty, invalid, or given. For example:
```

Comment:
What happens if I revert to version `kitten` Oo. Will it say `revert complete.` without error message??